dynamic_programming/word_break.py

```diff
@@ -0,0 +1,25 @@
+def wordbreak(word: str, word_dict: list) -> bool:
+    """
+    word ---> Input word
+    word_dict ---> Input dictionary
+    >>> wordbreak('deepImpact',['deep','Im','pact'])
+    True
+    >>> wordbreak('mangoinmango',['mango','in'])
+    True
+    >>> wordbreak('deepfacts',['deeps','fact'])
+    False
+    """
+    dict_set = set(word_dict)
+    start = [0]
+    # start[0] is the equivalent of a base-case
+    # from the recursive solution and start[-1] is the
+    # overall solution to the complete problem
+    for i in range(len(word)):
+        for j in start:
+            if word[j : i + 1] in dict_set:
+                start.append(i + 1)  # start of next word in dict

```

Comment:
```suggestion
    for i in range(1, len(word) + 1):
        for j in start:
            if word[j : i] in dict_set:
                start.append(i)  # start of next word in dict
```
No need to add 1 to the index every time, just shift the range of the for-loop.